uniffi_macros/src/export/metadata.rs

```diff
@@ -171,29 +154,13 @@ fn type_not_supported(ty: &impl ToTokens) -> syn::Error {
     )
 }
 
-fn write_json_metadata<T: Serialize>(path: &Utf8Path, val: &T) -> io::Result<TokenStream> {
-    // The usage of a `NamedTempFile` ensures that the file ends up in a consistent state if
-    // multiple processes or threads run this function in parallel with the same path (for example
-    // one `rustc` process and one `rust-analyzer` process).
-    //
-    // Without this, two serializers wanting to write to the same file could have their outputs
-    // merged together by interleaved writes. By first writing to a temporary file that is then
-    // atomically renamed to the desired file name, this is avoided.
-    let file = NamedTempFile::new_in(
-        path.parent()
-            .expect("paths passed to write_json_metadata must have a parent"),
-    )?;
-    serde_json::to_writer_pretty(&file, val)?;
-    file.persist(path)?;
-
-    // This configures the compiler to re-run this macro invocation when the file at `path` is
-    // changed (practically this should only happen if it is deleted).
-    //
-    // Ideally this function would return `io::Result<()>` and use a proc_macro function for
-    // "tracking" the file instead of returning this TokenStream. That API is still nightly-only
-    // and subject to change though, see https://github.com/rust-lang/rust/issues/73921
-    let path = path.as_str();
-    Ok(quote! {
-        const _: &str = ::std::include_str!(#path);
-    })
+fn create_metadata_static_var<T: Into<Metadata>>(name: impl IdentFragment, val: T) -> TokenStream {
```

Comment:
Why is this generic? I think it would be a bit easier to grok if any necessary `.into()` calls happened at the call site.